sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -606,20 +609,28 @@ public void setSessionState(String sessionId, byte[] sessionState) {
     @Override
     public void close() {
         asyncClient.close();
+
+        if (messageProcessor != null) {
+            messageProcessor.onComplete();
+        }
     }
 
     /**
-     * Given an {@code emitter}, queues that work in {@link SynchronousMessageSubscriber}. If the synchronous job has
-     * not been created, will initialise it.
+     * Given an {@code emitter}, creates a {@link SynchronousMessageSubscriber} to receive messages from Service Bus.
      */
-    private void queueWork(int maximumMessageCount, Duration maxWaitTime,
+    private synchronized void queueWork(int maximumMessageCount, Duration maxWaitTime,
         FluxSink<ServiceBusReceivedMessageContext> emitter) {
```

Comment:
This is because if multiple client are calling receive, we need to ensure that short lived sync subscriber is thread safe.

Reply:
You don't need an atomic variable then.

Reply:
You can also synchronize on a `lock` object rather than on an entire class to scope down how large this critical path is.

Reply:
yes: atomic is not needed  but our Checks during build time complains about it or I had to supress the warning .